### PR TITLE
feat(intellij): orchestrate AutoBot codegen

### LINKED ISSUE #5275: Add AutoBot scenario-to-recording codegen workflow
## User Scenarios & Testing
As an IntelliJ user, I can submit `/codegen <scenario>`, record the browser journey, review the proposed source changes, then generate, replay, and heal the scenario.

## Functional Requirements
- AutoBot local CLI owns capture, generation, replay, healing, and concise final reporting.
- The panel owns visible progress, cancellation, disposal, and source-edit confirmation.
- Reuse page objects/test classes and SHAFT locator guidance before creating files.
- Default to one replay-heal attempt; require explicit user wording for more.

## Edge Cases
Missing target URL; unavailable local CLI/MCP capability; cancellation; failed recording; no generated source; no Allure file.

## Success Criteria
Focused plugin tests prove state transitions, consent gate, one-heal ceiling, and summary/Allure result behavior.

## Assumptions
Recording starts from scenario or configured target URL; source edits and replay wait for explicit confirmation.

### PR DESCRIPTION
**Pull request by MohabMohie**
## Summary

- route free-text `/codegen <scenario>` to a Codex AutoBot workflow instead of plugin-owned browser capture
- coordinate `RECORD`, URL fallback, edit confirmation, `GENERATE`, `REPLAY`, one default `HEAL` retry, and terminal status per chat session
- keep recording-JSON replay deterministic and preserve the plugin as a UI/lifecycle coordinator
- enforce a read-only browser/capture MCP allowlist before consent, then launch a separate source-mutable invocation only after explicit approval
- require the actual launched AutoBot prompts to use SHAFT authoring/page-object guidance, reuse existing owners, follow POM and locator priority, and report retry replay progress
- validate recording, phase, class, and optional Allure evidence before presenting success or failure

Closes #5275
Related to #5273

## Checks

- `shaft-intellij/gradlew.bat test --tests com.shaft.intellij.ui.AssistantCodegenWorkflowCoordinatorTest --tests com.shaft.intellij.ui.AssistantCommandTest --tests com.shaft.intellij.ui.AssistantCommandToolIndexRoutingTest --tests com.shaft.intellij.ui.AssistantCommandRoutingTest --tests com.shaft.intellij.ui.AssistantLocalAgentRunnerCommandTest --tests com.shaft.intellij.ui.AssistantLocalAgentRunnerVerboseStreamTest --tests com.shaft.intellij.ui.ShaftAssistantPanelDisposalTest --tests com.shaft.intellij.ui.ShaftPanelSetupTest -Djava.awt.headless=true -Dallure.automaticallyOpen=false --rerun-tasks` — 529 tests, 0 failures, 0 errors, 0 skipped
- focused Codacy cleanup run across coordinator, runner, routing, and panel disposal tests — 99 tests, 0 failures, 0 errors, 0 skipped
- focused launched-prompt and retry-progress run across five impacted suites — 101 tests, 0 failures, 0 errors, 0 skipped
- `shaft-intellij/gradlew.bat test --tests com.shaft.intellij.ui.ShaftPluginScreenshotRendererTest -Djava.awt.headless=true -Dallure.automaticallyOpen=false` — 8 tests, 0 failures, 0 errors, 1 expected skip; no component/layout change, so no screenshot artifact retained
- `git diff --check` — clean
- independent adversarial reviews for the workflow, static-analysis cleanup, and launched-prompt/retry contract — Spec PASS, Quality APPROVED, ZERO BLOCKING FINDINGS
- refreshed GitHub PR Gate, IntelliJ build, CodeQL, Codacy, dependency submission, and installer checks — terminal green or intentional skip

## Continuation

Head: `d13b83dcb3d0fb408eb5225d333cda72d05f10bf`

State: exact-head implementation, reviews, checks, merge authority, and feedback audit are complete. Companion documentation is green on ShaftHQ/shafthq.github.io#1001.

Blockers: none. A configured Codex plus SHAFT MCP/browser E2E was not run under the selected focused headless local scope.

Next action: merge this PR and verify the merged head, then merge the companion documentation PR.

### CODE REVIEW

**Review by MohabMohie — COMMENTED** [review]
Independent adversarial PR review for exact head 185cb714ca048b6ff0deb7cc058f6d8a1cdd6024. Scope: complete eight-file IntelliJ AutoBot implementation against issue #5275, including routing, RECORD tool boundary, consent-separated mutable run, runner metadata recovery, panel lifecycle, terminal evidence validation, and changed tests. The live PR contains exactly eight changed files and every remote blob matches the locally reviewed fix-round-3 state; base is d5ff6533de3962405a0b4731b4054f7d9ad04c00. Evidence reviewed: focused RED/GREEN receipts; balanced 529-test run with 0 failures, 0 errors, and 0 skipped; fix-round-3 coordinator plus panel-disposal XML with 34 tests, 0 failures, 0 errors, and 0 skipped; live Build IntelliJ plugin check succeeded. Bot findings were inspected: bounded 1-2 digit parsing makes the NumberFormatException warning non-applicable, reflection warnings are test-only, and remaining complexity/style notices do not identify wrong behavior. Spec PASS. Quality APPROVED. ZERO BLOCKING FINDINGS in the reviewed patch. Posted as COMMENT because the reviewer identity matches the PR author; this is not self-approval. Current remote delivery gates remain separate: CodeQL (java) failed during Maven setup with HTTP 429, and Codacy remains ACTION_REQUIRED.

**Review by MohabMohie — COMMENTED** [review]
Independent adversarial rereview for exact head `9f829f28872577fef481800575d292db3c678d97`.

The pushed second commit contains exactly the reviewed three-file Codacy cleanup delta. Its binary diff from `185cb714ca048b6ff0deb7cc058f6d8a1cdd6024` hashes to `90f6f4bcf27a10ac54bf2902b24d5a37cebb4997`. The other five files from the original eight-file implementation retain their exact reviewed blobs.

Scope reviewed: eight narrow, line-local `NOPMD` comments on test reflection sites; behavior-preserving NPath extraction for routing, terminal validation, and runner metadata validation; reuse of the existing `Collectors` import; and removal of the redundant explicit public constructor while preserving a public implicit constructor. The bounded 1-2 digit heal-retry regex keeps the existing `Integer.parseInt` call safe, so the open NumberFormatException bot warning remains non-applicable.

Focused local XML evidence: 99 tests, 0 failures, 0 errors, and 0 skipped across coordinator, runner command, command routing, tool-index routing, and panel-disposal suites.

Spec PASS. Quality APPROVED. ZERO BLOCKING FINDINGS.

File: shaft-intellij/src/main/java/com/shaft/intellij/ui/AssistantCodegenWorkflowCoordinator.java [hunk 1]
```diff
@@ -0,0 +1,676 @@
...
+            return false;
+        }
+    }
+
+    private static int healRetryAllowance(String scenario) {
+        Matcher matcher = HEAL_RETRY_OVERRIDE.matcher(scenario == null ? "" : scenario);
+        if (!matcher.find()) {
+            return 1;
+        }
+        int requested = Integer.parseInt(matcher.group(1));
```
[the remark this answers is not in the document]
Comment (MohabMohie) [reply shaft-intellij/src/main/java/com/shaft/intellij/ui/AssistantCodegenWorkflowCoordinator.java:604]: At head `9f829f2`, `HEAL_RETRY_OVERRIDE` retains capture group `(\d{1,2})`. With Java regex defaults used here, this accepts only one or two decimal digits, so `matcher.group(1)` parses to a value from 0 through 99. `Integer.parseInt` therefore cannot receive malformed or overflow-sized input from this matcher. The cleanup only extracted complexity and preserved this behavior; no catch is needed.

**Review by MohabMohie — COMMENTED** [review]
Independent adversarial PR rereview for exact head `d13b83dcb3d0fb408eb5225d333cda72d05f10bf`.

The final commit has parent `9f829f28872577fef481800575d292db3c678d97`, changes exactly `AssistantCodegenWorkflowCoordinator.java` and `AssistantCodegenWorkflowCoordinatorTest.java`, and its binary delta hashes to the frozen reviewed value `40eb4d4c69a02981900a5d021e89ff78c0a1631c`. Earlier commits `185cb714ca048b6ff0deb7cc058f6d8a1cdd6024` and `9f829f28872577fef481800575d292db3c678d97` remain unchanged in the parent chain.

The launched RECORD and mutable prompts now name the real SHAFT recording, automated-test-authoring, and page-object skills; retain SHAFT MCP/CLI ownership, existing-owner inspection and `shaft_coding_partner_plan`, POM and locator priority, one replay plus bounded healing, read-only RECORD, consent separation, lifecycle, and terminal evidence. Progress tracking permits the required `GENERATE`, `REPLAY`, `HEAL`, `REPLAY` sequence while counting HEAL entries against `healRetryAllowance`: the default second loop is rejected and an explicit allowance permits additional loops.

Focused XML evidence across five impacted suites: 101 tests, 0 failures, 0 errors, and 0 skipped. The existing NumberFormatException thread is resolved, Codacy reports 0 issues, and current pending remote checks remain delivery evidence separate from this code-review verdict.

Spec PASS. Quality APPROVED. ZERO BLOCKING FINDINGS.

Posted as COMMENT because reviewer identity matches PR author; this is not self-approval.

**Comment by MohabMohie** [comment]
Disposition: false positive. The bounded retry regex accepts only one or two decimal digits, so the parsed value is 0-99; the proof reply is posted and the thread is resolved.

**Comment by MohabMohie** [comment]
Disposition: valid. The first exact-head adversarial review was addressed by the later static-analysis and launched-prompt fix rounds.

**Comment by MohabMohie** [comment]
Disposition: valid. The Codacy cleanup review passed; its exact three-file delta remains in the final head.

**Comment by MohabMohie** [comment]
Disposition: valid. The final launched-prompt and retry-progress review passed at the current exact head with 101 focused tests green.

**Comment by MohabMohie** [comment]
Disposition: valid. The original Codacy findings were fixed in the reviewed cleanup commit; the current Codacy check reports zero issues.

**Comment by MohabMohie** [comment]
Disposition: false positive for this patch. This is a GitHub-hosted runner warning about the third-party Maven dependency-submission action runtime, not a finding in the changed IntelliJ code; dependency churn is outside this scoped fix.

**Comment by MohabMohie** [comment]
Disposition: false positive for this patch. This is CodeQL manual-build overlay fallback telemetry; the exact-head Java analyses completed successfully and reported no code-scanning alert.
